sqlx-core/src/pool/metrics.rs

```diff
@@ -0,0 +1,85 @@
+//! Pool metric / event instrumentation.
+
+use std::{ops::Deref, sync::Arc, time::Duration};
+
+/// An observer of [`Pool`] metrics.
+///
+/// ```rust
```

Comment:
I didn't hit any issue with running this locally, but perhaps it should be `no_run`?

Reply:
There's nothing _to_ run, all the imperative code is in a function that's never called.